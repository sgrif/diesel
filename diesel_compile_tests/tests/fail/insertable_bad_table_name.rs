#[macro_use]
extern crate diesel;

table! {
    users {
        id -> Integer,
    }
}

#[derive(Insertable)]
#[diesel(table_name = users)]
struct UserOk {
    id: i32,
}

#[derive(Insertable)]
#[diesel(table_name(users))]
struct UserWarn {
    id: i32,
}

#[derive(Insertable)]
#[diesel(table_name)]
struct UserError1 {
    id: i32,
}

#[derive(Insertable)]
#[diesel(table_name = true)]
struct UserError2 {
    id: i32,
}

#[derive(Insertable)]
#[diesel(table_name = "")]
struct UserError3 {
    id: i32,
}

#[derive(Insertable)]
#[diesel(table_name = "not a path")]
struct UserError4 {
    id: i32,
}

#[derive(Insertable)]
#[diesel(table_name = does::not::exist)]
struct UserError5 {
    id: i32,
}

fn main() {}
