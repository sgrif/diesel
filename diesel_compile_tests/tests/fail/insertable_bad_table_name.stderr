error: `table_name` is either unexpected or has bad format
  --> $DIR/insertable_bad_table_name.rs:17:10
   |
17 | #[diesel(table_name(users))]
   |          ^^^^^^^^^^

error: `table_name` is either unexpected or has bad format
  --> $DIR/insertable_bad_table_name.rs:23:10
   |
23 | #[diesel(table_name)]
   |          ^^^^^^^^^^

error: expected identifier
  --> $DIR/insertable_bad_table_name.rs:29:23
   |
29 | #[diesel(table_name = true)]
   |                       ^^^^

error: expected identifier
  --> $DIR/insertable_bad_table_name.rs:35:23
   |
35 | #[diesel(table_name = "")]
   |                       ^^

error: expected identifier
  --> $DIR/insertable_bad_table_name.rs:41:23
   |
41 | #[diesel(table_name = "not a path")]
   |                       ^^^^^^^^^^^^

error[E0433]: failed to resolve: use of undeclared crate or module `does`
  --> $DIR/insertable_bad_table_name.rs:47:23
   |
47 | #[diesel(table_name = does::not::exist)]
   |                       ^^^^ use of undeclared crate or module `does`
